pkg/codegen/java/gen_program.go

```diff
@@ -528,6 +528,29 @@ func pulumiImport(pkg string, module string, member string) string {
 	return "com.pulumi." + sanitizeImport(pkg) + "." + sanitizeImport(module) + "." + member
 }
 
+func pulumiResourceImport(r *pcl.Resource, pkg string, module string, member string) string {
+	module = cleanModule(module)
+	if pkg == "pulumi" && module == "pulumi" && (member == "StackReference" || member == "StackReferenceArgs") {
+		return "com.pulumi.resources." + member
+	}
```

Comment:
I wonder if this is worth hoisting out to something like `ifIsBuiltin` or the like, since we use it above as well.